### ISSUE TITLE
conv2dTranspose gives different resutls on WebGL and WASM backends

### ISSUE THREAD

**Issue by drnemor** [tensorflow/tfjs#4892]
**System information**
- TensorFlow.js version: 3.2
- Browser version: Chrome 89 / pop! OS 20.04 LTS

Output of the [tf.layers.conv2dTranspose](https://js.tensorflow.org/api/latest/#layers.conv2dTranspose) on WebGL backend is different from WASM backend even for simple cases. Other layers like conv2d, upsample, etc work fine (with very small numerical differences).

Simple reproduction code:
```javascript
var cfg = {
    filters: 1,
    kernelSize: 2, 
    strides: 2, 
    padding: "same", 
    batchInputShape: [1, 144 , 256, 3],
    useBias: false,
    kernelInitializer: 'ones'
}

async function main() {
    await tf.setBackend('webgl');
    const data_from_webgl = await evaluateModel();

    await tf.setBackend('wasm');
    const data_from_wasm = await evaluateModel();

    console.log(data_from_webgl);
    console.log(data_from_wasm);
}

async function evaluateModel() {
    const model = tf.sequential();
    model.add(tf.layers.conv2dTranspose(cfg));
    const prediction = tf.tidy(() => {
        const imgTensor = tf.randomNormal([1, 144, 256, 3], 0, 1, 'float32', 11111);
        const y = model.predict(imgTensor);
        return y;
    });
    return prediction.data();
}

main();
```

**Comment by pyu10055** [comment]
@drnemor I am curious if you are using tf.randomNormal, the input would change on every execution, it will lead to different output.

**Comment by drnemor** [comment]
@pyu10055 I used the same random seed, so randomNormal gives same tensor on each execution regardless of chosen backend. You can check that by replacing ```model.add(tf.layers.conv2dTranspose(cfg)) ``` with ```model.add(tf.layers.activation({batchInputShape: [1, 144 , 256, 3], activation: 'linear'}))```. This gives exactly the same tensors on both backends. Also, you can check other layers like ```tf.layers.conv2d```, they also produce the same results, but not ```conv2dTranspose```.
